### PR TITLE
Gun code override unshittening part like, two by this point

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

So I've got a hilarious tale for you. In order to make the few changes to gun code that we actually did, we were completely overriding the tg gun.dm file. For the uninitiated, this is NOT A GOOD THING.

So, I took the time to go through and make our changes massively more modular, including componentizing both gun safeties and the company examines thing.

There is one and only one small functionality loss that has occurred because of this. Fire selection from automatic to single fire, and from automatic to burst fire, does not exist anymore, due to technical limitations (I don't want to fix the awful way it was done).

Good news, if you want to do single fire on an automatic gun, just click once.
Good-er news, fire selection between burst and single fire already existed as a thing on tg before this, so yippie!

THERES STILL SOME IMPORTANT STUFF TO DO AS ALWAYS:

- [x] MAKE THE SAFETY TOGGLE BUTTON ACTUALLY WORK I FUCKED THAT UP A BIT
- [x] I might have to fix the little hud that pops up when you pull a gun out that's supposed to show your ammo but barely ever works
- [x] Some guns might have lost autofire without me realizing due to the sheer number of edits that had to be made. Of course its gonna take a good while to test every single skyrat gun we have, because there are a few.

## How This Contributes To The Skyrat Roleplay Experience

It's really so much better if we don't override entire code files when we don't need to.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

MORE THAN THIS SOON GIMME A BIT

image
image
image
image
image
image
image
image
image

  
</details>

## Changelog

:cl:
code: We no longer completely override a few tg files, including the automatic fire component, and the gun file.
refactor: Gun safeties and manufacturer names have been refactored into their own components.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/company_imports/code/objects/nri_surplus/ak25.dm [hunk 1]
```diff
@@ -21,10 +21,12 @@
 	load_empty_sound = 'modular_skyrat/modules/gunsgalore/sound/guns/interact/ltrifle_magin.ogg'
 	eject_sound = 'modular_skyrat/modules/gunsgalore/sound/guns/interact/ltrifle_magout.ogg'
 	alt_icons = TRUE
-	company_flag = COMPANY_IZHEVSK
 	spread = 29
 	recoil = 0.1
 
+/obj/item/gun/ballistic/automatic/ak25/give_manufacturer_examine()
+	AddComponent(/datum/component/manufacturer_examine, COMPANY_IZHEVSK)
```
Comment (Zonespace27) [inline modular_skyrat/modules/company_imports/code/objects/nri_surplus/ak25.dm:28]: Could be an element instead

Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/objects/nri_surplus/ak25.dm:28]: It could, theoretically speaking, be an element, however.

Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/objects/nri_surplus/ak25.dm:28]: 1100 line long sec_haul gun file.

**Comment by Zenitheevee** [comment]
We suffer for our hubris. God speed.

File: modular_skyrat/modules/modular_weapons/code/automatic.dm [hunk 2]
```diff
@@ -51,7 +51,9 @@
 	fire_sound = 'sound/weapons/gun/smg/shot_alt.ogg'
 	weapon_weight = WEAPON_MEDIUM
```
Comment (RatFromTheJungle) [inline modular_skyrat/modules/modular_weapons/code/automatic.dm:51]: this is supposed to be full auto, the burst fire was a secondary option (as burst is much-much stronger)

File: modular_skyrat/modules/modular_weapons/code/automatic.dm [hunk 3]
```diff
@@ -51,7 +51,9 @@
...
 	w_class = WEIGHT_CLASS_BULKY
-	company_flag = COMPANY_CANTALAN
+
+/obj/item/gun/ballistic/automatic/cfa_wildcat/give_manufacturer_examine()
+	AddComponent(/datum/component/manufacturer_examine, COMPANY_CANTALAN)
 
 /obj/item/gun/ballistic/automatic/cfa_wildcat/no_mag
```
Comment (RatFromTheJungle) [inline modular_skyrat/modules/modular_weapons/code/automatic.dm:62]: (ditto) also supposed to be automatic as a baseline, with burst as the secondary, and if you're getting rid of one, it should default to full auto

File: modular_skyrat/modules/sec_haul/code/guns/guns.dm [hunk 4]
```diff
@@ -206,7 +213,8 @@
 	burst_size = 3
 	fire_delay = 2
 	spread = 9
-	company_flag = COMPANY_ARMADYNE
+
+/obj/item/gun/ballistic/
```
Comment (RatFromTheJungle) [inline modular_skyrat/modules/sec_haul/code/guns/guns.dm]: ?

Comment (Paxilmaniac) [reply modular_skyrat/modules/sec_haul/code/guns/guns.dm]: good question

File: _maps/RandomZLevels/blackmesa.dmm [hunk 5]
```diff
@@ -3998,9 +3998,7 @@
 /area/awaymission/black_mesa/security_outpost)
 "aYe" = (
 /obj/structure/rack/gunrack,
-/obj/item/gun/ballistic/revolver/mateba{
-	fire_select_modes = list(1,2)
```
Comment (ghost) [inline _maps/RandomZLevels/blackmesa.dmm:4002]: Who the fuck did this.

**Review by ghost — COMMENTED** [review]
Empty icon_state `mp5_burst` in `modular_skyrat/modules/gunsgalore/icons/guns/gunsgalore_guns40x32.dmi (0) `

Code checks out from what I understand of it

But-

File: _maps/RandomZLevels/blackmesa.dmm (hunk 5 above)
Comment (Paxilmaniac) [reply _maps/RandomZLevels/blackmesa.dmm:4002]: dont ever trust no black mesa devs

**Comment by Paxilmaniac** [comment]
free merge conflict, just a week away

**Review by tf-4 — APPROVED** [review]

**Pull request merged**